cmake/kokkos_arch.cmake

```
@@ -1213,8 +1213,8 @@ if(KOKKOS_ENABLE_HIP AND NOT AMDGPU_ARCH_ALREADY_SPECIFIED AND NOT KOKKOS_IMPL_A
   else()
     execute_process(COMMAND ${ROCM_ENUMERATOR} OUTPUT_VARIABLE GPU_ARCHS)
     string(LENGTH "${GPU_ARCHS}" len_str)
-    # enumerator always output gfx000 as the first line
-    if(${len_str} LESS 8)
+    # Exits early if no GPU was detected
+    if(${len_str} EQUAL 0)
```

```suggestion
    # Exits early if no GPU was detected
    if("${GPU_ARCHS}" STREQUAL "")
```
would be more concise since we only care about the string being empty and matches line 1238.